dev-mode/main.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/signal"
+	"path"
+	"serverless/dev-mode-extension/agent"
+	"serverless/dev-mode-extension/extension"
+	"serverless/dev-mode-extension/logsapi"
+	"strings"
+	"syscall"
+
+	"github.com/golang-collections/go-datastructures/queue"
+	log "github.com/sirupsen/logrus"
+)
+
+// INITIAL_QUEUE_SIZE is the initial size set for the synchronous logQueue
+const INITIAL_QUEUE_SIZE = 5
+
+func main() {
+	extensionName := path.Base(os.Args[0])
+	printPrefix := fmt.Sprintf("[%s]", extensionName)
+	logger := log.WithFields(log.Fields{"agent": extensionName})
+
+	extensionClient := extension.NewClient(os.Getenv("AWS_LAMBDA_RUNTIME_API"))
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGTERM, syscall.SIGINT)
+	go func() {
+		s := <-sigs
+		cancel()
+		logger.Info(printPrefix, "Received", s)
+		logger.Info(printPrefix, "Exiting")
+	}()
+
+	// Register extension as soon as possible
+	_, err := extensionClient.Register(ctx, extensionName)
+	if err != nil {
+		panic(err)
+	}
+
+	// A synchronous queue that is used to put logs from the goroutine (producer)
+	// and process the logs from main goroutine (consumer)
+	logQueue := queue.New(INITIAL_QUEUE_SIZE)
+	// Helper function to empty the log queue
+	var logsStr string = ""
+	var requestId string = ""
+	flushLogQueue := func(force bool) {
+		for !(logQueue.Empty() && (force || strings.Contains(logsStr, string(logsapi.RuntimeDone)))) {
+			logs, err := logQueue.Get(1)
+			if err != nil {
+				logger.Error(printPrefix, err)
+				return
+			}
+			logsStr = fmt.Sprintf("%v", logs[0])
```

Comment:
I think `%v` should be used only for debug. It seems you should actually be parsing the log in its json format to get log type and flag as runtime done when it is done, like in https://github.com/serverless/console/blob/bfb77d9848e8fc267443afbe97e6e5f4dffa1140/go/packages/aws-lambda-otel-extension/external/reporter/parser_logs.go#L153
I know it should be pretty rare, but any platform runtimedone word in the same way inside log may wrongly flag as done
Also you are running a string compare check in the entire JSON log response from AWS for something that should be a .type field

Reply:
I am going to hold off on parsing logs for now. I think the requirements for log parsing may have changed a bit since we were last building this layer. I will for sure keep this in mind when we add logs parsing back in 👍

Reply:
Going to resolve this for now 👍 I will add more log parsing if we end up changing the schema. Thanks for your help with the benchmark stuff!